WcaOnRails/app/models/competition.rb

```diff
@@ -1172,37 +1172,41 @@ def schedule_wcif
     }
   end
 
-  def set_wcif_events!(wcif_events, current_user)
-    events_schema = { "type" => "array", "items" => CompetitionEvent.wcif_json_schema }
-    JSON::Validator.validate!(events_schema, wcif_events)
-
+  def set_wcif!(wcif, current_user)
+    JSON::Validator.validate!(Competition.wcif_json_schema, wcif)
     ActiveRecord::Base.transaction do
-      # Remove extra events.
-      self.competition_events.each do |competition_event|
-        wcif_event = wcif_events.find { |e| e["id"] == competition_event.event.id }
-        event_to_be_removed = !wcif_event || !wcif_event["rounds"]
-        if event_to_be_removed
-          raise WcaExceptions::BadApiParameter.new("Cannot remove events from a confirmed competition") unless current_user.can_add_and_remove_events?(self)
-          competition_event.destroy!
-        end
+      update_persons_wcif!(wcif["persons"], current_user) if wcif["persons"]
+      set_wcif_events!(wcif["events"], current_user) if wcif["events"]
+      set_wcif_schedule!(wcif["schedule"], current_user) if wcif["schedule"]
+    end
+  end
+
+  def set_wcif_events!(wcif_events, current_user)
```

Comment:
Would it make sense to make these three helper methods `private` now?

Reply:
Haha, that was my exact intuition, but we do have a lot of tests for each of these methods and I don't know any way of testing private methods.